### PR TITLE
Exterminate FakeThread with extreme prejudice

### PR DESCRIPTION
**Pull request by gallantron**
There's currently a contraption called `FakeThread` in `exe2minixfs` to enable natively compiling outside the context of SWEB.

It's also really ugly and unnecessary. So I'm replacing it with a `getcwd()` method.

### CODE REVIEW

File: common/source/kernel/Thread.cpp [hunk 1]
```diff
@@ -108,6 +108,13 @@ FileSystemInfo* Thread::getWorkingDirInfo(void)
   return working_dir_;
 }
 
+FileSystemInfo* getcwd()
+{
+  if (FileSystemInfo* info = currentThread->getWorkingDirInfo())
```
Comment (dgruss) [inline common/source/kernel/Thread.cpp]: what if currentThread == 0? where do you catch that?

Comment (gallantron) [reply common/source/kernel/Thread.cpp]: isn't that impossible? `currentThread` is only ever set from `Scheduler`'s `threads_` queue, which should never contain `nullptr`.

The old code had the `nullptr` check because `FakeThread* currentThread;` was set to it.

File: userspace/libc/src/printf.c [hunk 2]
```diff
@@ -495,7 +495,7 @@ extern int printf(const char *format, ...)
       if(*format == '.') {
         ++format; 
         char num[4];
-        c = 0;
+        size_t c = 0;
```
Comment (gallantron) [inline userspace/libc/src/printf.c]: _(note: unrelated build fix after #138, which broke this variable reuse due to scoping change)_
